theano/compile/function.py

```diff
@@ -267,4 +295,16 @@ def opt_log1p(node):
     # We need to add the flag check_aliased inputs if we have any mutable or
     # borrowed used defined inputs
     fn._check_for_aliased_inputs = check_for_aliased_inputs
+
+    if outputs_dict_format:
+
+        fn_dict_output = (lambda *args, **kwargs: output_dictionary_wrapper(args, kwargs, fn = fn, keys = output_keys))
```

Comment:
Thanks.  I see where I can put it into Function.call, but adding it will require additional arguments for the Function class.  

So the current header: 

def __init__(self, fn, input_storage, output_storage, indices, outputs, defaults, unpack_single, return_none, maker):

would become: 

def __init__(self, fn, input_storage, output_storage, indices, outputs, defaults, unpack_single, return_none, maker,output_to_dictionary,output_keys):

I'll code and commit this change just to see that it works, but I imagine that adding too much configuration to the constructor arguments could hurt readability.